File: packages/authentication/ios/Plugin/Classes/Options/SendEmailVerificationOptions.swift
import Foundation
import Capacitor
import FirebaseAuth

@objc public class SendEmailVerificationOptions: NSObject {
    private var actionCodeSettings: ActionCodeSettings?

    init(actionCodeSettings: JSObject?) {
        self.actionCodeSettings = FirebaseAuthenticationHelper.createActionCodeSettings(actionCodeSettings)
    }

    func getActionCodeSettings() -> ActionCodeSettings? {
        return actionCodeSettings
    }
}

File: packages/authentication/android/src/main/java/app/capgo/capacitor/firebase/authentication/SendEmailVerificationOptions.java
package app.capgo.capacitor.firebase.authentication.classes.options;

import androidx.annotation.Nullable;
import app.capgo.capacitor.firebase.authentication.FirebaseAuthenticationHelper;
import com.getcapacitor.JSObject;
import com.google.firebase.auth.ActionCodeSettings;
import org.json.JSONException;

public class SendEmailVerificationOptions {

    @Nullable
    private ActionCodeSettings actionCodeSettings;

    public SendEmailVerificationOptions(@Nullable JSObject actionCodeSettings) throws JSONException {
        this.actionCodeSettings = FirebaseAuthenticationHelper.createActionCodeSettings(actionCodeSettings);
    }

    @Nullable
    public ActionCodeSettings getActionCodeSettings() {
        return actionCodeSettings;
    }
}
